metricbeat/module/system/raid/raid_test.go

```diff
@@ -34,6 +72,12 @@ func TestData(t *testing.T) {
 }
 
 func TestFetch(t *testing.T) {
+
+	//override the var in ioctl.go for testing
+	makeNewDevice = func(dev string) (MDData, error) {
+		return MockData(" "), nil
+	}
```

Comment:
You can do this in `init()` so it is done only once and you don't have risks of race conditions if these tests happen to be run in parallel or with the race detector.

Reply:
I'm not sure what you mean? The `init()` function in `raid.go` ? Can that be changed in the test somehow?

Reply:
Ah, you can have a separate `init()` in the test file, it looks like?

Reply:
Yes, there can be multiple `init()` in the same package.